bndtools.core/src/org/bndtools/core/editors/quickfix/AddBundleCompletionProposal.java

```diff
@@ -0,0 +1,142 @@
+import bndtools.Plugin;
+import bndtools.central.Central;
+
+class AddBundleCompletionProposal extends WorkspaceJob implements IJavaCompletionProposal {
+
+	final BundleId				bundle;
+	final String				displayString;
+	final int					relevance;
+	final IInvocationContext	context;
+	final Project				project;
+	final String				pathtype;
+	final String				fqn;
+
+	public AddBundleCompletionProposal(String fqn, BundleId bundle, int relevance, IInvocationContext context,
+		Project project, String pathtype) {
+		super("Adding '" + bundle + "' to " + project + " " + pathtype);
+		this.fqn = fqn;
+		this.bundle = bundle;
+		this.relevance = relevance;
+		this.context = context;
+		this.project = project;
+		this.pathtype = pathtype;
+		this.displayString = Strings.format("Add %s %s to %s (found %s)", bundle.getBsn(), bundle.getShortVersion(),
+			pathtype, fqn);
+	}
+
+	@Override
+	public void apply(org.eclipse.jface.text.IDocument document) {
+		schedule();
+	}
+
+	/**
+	 * @see org.eclipse.jface.text.contentassist.ICompletionProposal#getSelection(org.eclipse.jface.text.IDocument)
+	 */
+	@Override
+	public Point getSelection(org.eclipse.jface.text.IDocument document) {
+		return new Point(context.getSelectionOffset(), context.getSelectionLength());
+	}
+
+	@Override
+	public String getAdditionalProposalInfo() {
+		return displayString;
+	}
+
+	@Override
+	public String getDisplayString() {
+		return displayString;
+	}
+
+	@Override
+	public Image getImage() {
+		return Icons.image("bundle");
+	}
+
+	@Override
+	public IContextInformation getContextInformation() {
+		return new IContextInformation() {
+
+			@Override
+			public String getContextDisplayString() {
+				return "Added " + bundle + " to build path";
+			}
+
+			@Override
+			public Image getImage() {
+				return Icons.image("bundle");
+			}
+
+			@Override
+			public String getInformationDisplayString() {
+				return "Added " + bundle + " to build path - info";
+			}
+
+		};
+	}
+
+	@Override
+	public int getRelevance() {
+		return relevance;
+	}
+
+	@Override
+	public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
+		try {
+			IStatus status = Central.bndCall(() -> {
+				BndEditModel model = new BndEditModel(project);
+				model.load();
+
+				switch (pathtype) {
+					case Constants.TESTPATH :
+						model.addPath(bundle, Constants.TESTPATH);
+						break;
+
+					case Constants.BUILDPATH :
+					default :
+						model.addPath(bundle, Constants.BUILDPATH);
+						break;
+				}
+				model.saveChanges();
+				Central.refreshFile(project.getPropertiesFile());
+				return Status.OK_STATUS;
+			});
+
+			String[] determine = Descriptors.determine(fqn)
+				.unwrap();
+
+			assert determine[0] != null : "We must have found a package";
+
+			if (determine[1] == null) {
+				context.getCompilationUnit()
+					.createImport(fqn + ".*", null, monitor);
+			} else {
+				context.getCompilationUnit()
+					.createImport(fqn, null, monitor);
+			}
+			return status;
```

Comment:
- I think duplicates are automatically skipped, at least I read something in the comments
- I just cannot find a definition of the error codes. 

I tested this in the UI and it seems to work fine so far. Shall we wait until we have actually test cases. I could not find a good definition of the error codes, what arguments they supplied, and what cases they were reported. I basically could only find import errors.

Having test code would allow us to see each case we want to support. I reasoned this was the safe way and the Eclipse quick fixes support cleaning up if necessary.

I am a big fan of less is more. I'd like to start very simple that covers 80%, and then add the missing cases explicitly.